content/doc/developer/internationalization/new-internationalization.adoc

```diff
@@ -0,0 +1,339 @@
+|===
+a|
+....
+<h1>${%Path to file (.ipa or .apk)}</h1>
+....
+
+|===
+
+You need to create two separate strings in this case:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<h1>${%Path to file} (${%.ipa or .apk})</h1>
+....
+
+|===
+
+=== Creation of html files for help tips.
+
+It is possible in jelly files to configure entries which can show a tip
+message when the user clicks on a help icon.
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+<f:entry title="${%title}" help="/plugin/myPlugin/help.html">
+....
+
+|===
+
+Then, just create the file help.html for English or help_xx.html for the
+'xx' language.
+
+== What translators need to know/do
+
+The Jenkins project always welcomes contributions to translations. If
+you are interested in helping us, just file a pull request on GitHub. In
+the remainder of this section, we'll discuss what needs to be translated
+and how.
+
+[cols="",]
+|===
+|Remember that properties files must be encoded in
+http://java.sun.com/javase/6/docs/api/java/util/Properties.html[ISO
+8859-1]. This should be the default anyway.
+|===
+
+=== Translation tools
+
+* https://wiki.jenkins.io/display/JENKINS/Translation+Tool[Translation
+Tool] : Command line tool to help translators to generate files, add
+missing keys, and more.
+* https://wiki.jenkins.io/display/JENKINS/Translation+Assistance+Plugin[Translation
+Assistance Plugin] : Adds a dialog box in Jenkins to translate and send
+missing keys.
+
+=== Translating `+Messages.properties+`
+
+Developers place messages that require localizations in
+`+Messages.properties+`. Those need to be translated in the usual
+manner. See `+Messages_ja.properties+` in the core as an example if you
+are new to this process.
+
+Sometimes looking at `+Messages.properties+` alone doesn't give you
+enough contextual information as to where the messages are used. For
+this, developers are encouraged to access messages by using the
+type-safe `+Messages+` class generated by
+http://java.net/projects/localizer/[localizer]. To find out where
+messages are actually used, use your IDE to find all the usages of the
+message format method.
+
+=== Translating message references in Jelly
+
+The other messages that need to be translated are in Jelly view files,
+which are in `+src/main/resources/**.jelly+`. To localize them, first
+you run Maven to generate skeleton property file for your locale:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+$ cd jenkins/core  (or a plugin dir)
+$ mvn stapler:i18n -Dlocale=fr
+....
+
+|===
+
+This will generate a bunch of `+*_fr.properties+` all over
+`+src/main/resources+` with an empty value. If the file already exists,
+it will append missing entries to existing files.
+
+You then need to work on each such property file and translate messages.
+You don't have to translate the entire file — if you leave some entries
+empty, they'll fall back to the default locale.
+
+[cols="",]
+|===
+|https://addons.mozilla.org/en-US/firefox/addon/1333[Quick locale
```

Comment:
```suggestion
|https://https://addons.mozilla.org/en-US/firefox/addon/quick-locale-switcher2/[Quick Locale Switcher]
```
The existing link is broken.  I went to the Firefox extensions list and found another addon with the same title so substituted that URL.  Please verify that this is correct.